src/CoreWCF.Primitives/src/CoreWCF/IdentityModel/Tokens/SessionSecurityTokenHandler.cs

```diff
@@ -0,0 +1,733 @@
+        /// Creates a security token based on a token descriptor.
+        /// </summary>
+        /// <param name=nameof(tokenDescriptor)>The token descriptor.</param>
+        /// <returns>A security token.</returns>
+        /// <exception cref="ArgumentNullException">Thrown if 'tokenDescriptor' is null.</exception>
+        public override SecurityToken CreateToken(SecurityTokenDescriptor tokenDescriptor)
+        {
+            if (null == tokenDescriptor)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull(nameof(tokenDescriptor));
+            }
+
+            if (Configuration == null)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ID4272)));
+            }
+
+            ClaimsPrincipal principal = new ClaimsPrincipal(tokenDescriptor.Subject);
+
+            if (Configuration.SaveBootstrapContext)
+            {
+                SecurityTokenHandlerCollection bootstrapTokenCollection = CreateBootstrapTokenHandlerCollection();
+                if (!bootstrapTokenCollection.CanWriteToken(tokenDescriptor.Token))
+                {
+                    throw DiagnosticUtility.ExceptionUtility.ThrowHelperError(new InvalidOperationException(SR.Format(SR.ID4010, tokenDescriptor.Token.GetType().ToString())));
+                }
+
+                (principal.Identities as ReadOnlyCollection<ClaimsIdentity>)[0].BootstrapContext = new BootstrapContext(tokenDescriptor.Token, bootstrapTokenCollection[tokenDescriptor.Token.GetType()]);
+            }
+
+            DateTime validFrom = (tokenDescriptor.Lifetime.Created.HasValue) ? (DateTime)tokenDescriptor.Lifetime.Created : DateTime.UtcNow;
+            DateTime validTo = (tokenDescriptor.Lifetime.Expires.HasValue) ? (DateTime)tokenDescriptor.Lifetime.Expires : DateTime.UtcNow + SessionSecurityTokenHandler.DefaultTokenLifetime;
+
+            return new SessionSecurityToken(principal, null, validFrom, validTo);
+        }
+
+        /// <summary>
+        /// Creates a <see cref="SessionSecurityToken"/> based on an <see cref="ClaimsPrincipal"/> and a valid time range.
+        /// </summary>
+        /// <param name="principal"><see cref="ClaimsPrincipal"/></param>
+        /// <param name="context">Caller defined context string</param>
+        /// <param name="endpointId">Identifier of the endpoint to which the token is scoped.</param>
+        /// <param name="validFrom">Earliest valid time.</param>
+        /// <param name="validTo">Latest valid time.</param>
+        public virtual SessionSecurityToken CreateSessionSecurityToken(
+            ClaimsPrincipal principal,
+            string context,
+            string endpointId,
+            DateTime validFrom,
+            DateTime validTo)
+        {
+            if (null == principal)
+            {
+                throw DiagnosticUtility.ExceptionUtility.ThrowHelperArgumentNull("principal");
+            }
```

Comment:
nameof. You can just search for the text `ThrowHelperArgumentNull("` to find instances of this.